### ISSUE TITLE
BUG: trying to upcast structs

### ISSUE THREAD

**Issue by paulcsmith** [crystal-lang/crystal#7481]
```
BUG: trying to upcast Dexter::Formatters::JsonLogFormatter <- Dexter::Formatters::BaseLogFormatter (Exception)
   from ???
   from ???
   from ???
```

Pull request with a Dockerfile that makes it easy to test and see what happens: https://github.com/luckyframework/avram/pull/56. Instructions there should make it easy to run and test out.

**The weird part is that [adding another struct that inherits from the parent makes the compiler work again](https://github.com/luckyframework/avram/pull/56/files#diff-d589fcc2891f9a552b3ed94627df66c0R3)** 🤔

The other weird part is that the Dexter library works just fine: https://github.com/luckyframework/dexter and so does Lucky, which uses Dexter: https://github.com/luckyframework/lucky/pull/698/files

I'd love to help debug this but there are no line numbers and I'm not really even sure where to start :S I'm guessing this has something to do with Struct inheritance. But there is only one parent class and one child so it seems like it should work fine (and it does except in this one weird case)

**Comment by paulcsmith** [comment]
It looks like this is fixed in Crystal 0.29 🎉https://github.com/luckyframework/avram/pull/56

**Comment by asterite** [comment]
Cool! 🎉 

> The weird part is that adding another struct that inherits from the parent makes the compiler work again 🤔

> It looks like this is fixed in Crystal 0.29

Indeed! That's what #7787 fixed 😄
